From 98e7a6ae2cc1c44ef7ef801ad3149f7a49b3c8fc Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Wed, 8 Apr 2020 17:48:12 -0300
Subject: [PATCH 159/171] Try again to fix Clang's -Wconstant-logical-operand
 warning

Commit daaa55725969aed87665c366d7cc96304c4e5f4c changed the enum to
bool, but the warning is still happening.

Change-Id: Ibdc95e9af7bd456a94ecfffd1603f332a356c788
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit e3a6ede895e1a2f59bc1b2947a1552c47681ed6c)
Reviewed-by: Lars Knoll <lars.knoll@qt.io>
---
 src/corelib/kernel/qmetatype_p.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/corelib/kernel/qmetatype_p.h b/src/corelib/kernel/qmetatype_p.h
index d743d5a5c7..50e5654221 100644
--- a/src/corelib/kernel/qmetatype_p.h
+++ b/src/corelib/kernel/qmetatype_p.h
@@ -108,6 +108,11 @@ public: \
 #define QT_DECLARE_WIDGETS_MODULE_TYPES_ITER(TypeName, TypeId, Name) \
     QT_ASSIGN_TYPE_TO_MODULE(Name, QModulesPrivate::Widgets);
 
+QT_WARNING_PUSH
+#if defined(Q_CC_CLANG) && Q_CC_CLANG >= 900
+QT_WARNING_DISABLE_CLANG("-Wconstant-logical-operand")
+#endif
+
 QT_FOR_EACH_STATIC_PRIMITIVE_TYPE(QT_DECLARE_CORE_MODULE_TYPES_ITER)
 QT_FOR_EACH_STATIC_PRIMITIVE_POINTER(QT_DECLARE_CORE_MODULE_TYPES_ITER)
 QT_FOR_EACH_STATIC_CORE_CLASS(QT_DECLARE_CORE_MODULE_TYPES_ITER)
@@ -115,6 +120,8 @@ QT_FOR_EACH_STATIC_CORE_POINTER(QT_DECLARE_CORE_MODULE_TYPES_ITER)
 QT_FOR_EACH_STATIC_CORE_TEMPLATE(QT_DECLARE_CORE_MODULE_TYPES_ITER)
 QT_FOR_EACH_STATIC_GUI_CLASS(QT_DECLARE_GUI_MODULE_TYPES_ITER)
 QT_FOR_EACH_STATIC_WIDGETS_CLASS(QT_DECLARE_WIDGETS_MODULE_TYPES_ITER)
+
+QT_WARNING_POP
 } // namespace QModulesPrivate
 
 #undef QT_DECLARE_CORE_MODULE_TYPES_ITER
-- 
2.31.1

